Comments cause warning messages

The comment style (/*! */) will cause a warning message:  
   
     Missing XML comment for publicly visible type or member. 
     But the alternative style (i.e. ///) will not cause any warning messages.

The xml special character "<, >" will cause a warning message too

    warning CS1570: XML comment on 'myclass' has badly formed XML -- 
    'A name was started with an invalid character.". A 

Need to escape them using `&lt;` and `&gt;`

--- a/Models/Plant/Organs/Leaf.cs
+++ b/Models/Plant/Organs/Leaf.cs
@@ -13,67 +13,66 @@ using APSIM.Shared.Utilities;
 
 namespace Models.PMF.Organs
 {
-    /*! <summary>
-        A generic leaf model
-        </summary>
-        \param KDead The extinction coefficient for dead leaf (\f$k_d\f$).
-        \param FrostFraction The fraction of leaf death caused by frost event.
-        \retval MaxCover The maximum coverage (\f$C_{max}\f$) with default value 1, 
-            which is set by manager sowing. 
-        \retval LAI Leaf area index for green leaf (\f$\text{LAI}_{g}\f$, \f$m^2 m^{-2}\f$)
-        \retval LAIDead Leaf area index for dead leaf  (\f$\text{LAI}_{d}\f$, \f$m^2 m^{-2}\f$)
-        \retval LAITotal Total LAI including live and dead parts (\f$m^2 m^{-2}\f$)
-            \f[
-                LAI = \text{LAI}_{g} + \text{LAI}_{d}
-            \f]
-        \retval CoverGreen Cover for green leaf (\f$C_g\f$, unitless). 
-            \f$C_g\f$ is calculated according to
-            extinction coefficient of green leaf (\f$k_{g}\f$).
-            \f[
-                C_{g}=C_{max}(1-\exp(-k_{g}\frac{\text{LAI}_{g}}{C_{max}}))
-            \f]
-            where, \f$k\f$ is the extinction coefficient which calculates 
-            by parameter "ExtinctionCoeff". 
-            As the default value of \f$C_{max}\f$ is 1, the function is reduced to
-            \f[
-                C_{g}=1-\exp(-k_{g}\text{LAI}_{g})
-            \f]
-        \retval CoverDead Cover for dead leaf (\f$C_d\f$, unitless).
-            \f$C_d\f$ is calculated according to
-            extinction coefficient of dead leaf (\f$k_{d}\f$).
-            \f[
-                C_{d}=1-\exp(-k_{d}\text{LAI}_{d})
-            \f]
-        \retval CoverTotal Total cover for green and dead leaves (\f$C_t\f$, unitless).
-            \f[
-                C_{t} = 1 - (1 - C_{g})(1 - C_{d})
-            \f]
-        
-        \retval Height Plant height (mm). 
-        \retval Depth Plant depth (mm). Equal to plant height (not sure its function?)
-        \retval FRGR Fractional relative growth rate (unitless, 0-1)
-            with 1.0 at full growth rate and 0.0 at no growth.
-        \retval PotentialEP Potential evapotranspiration. Set by MICROCLIMATE.
-        
-        <remarks>
-        The organ "Leaf" consists of a series of \ref LeafCohort "cohort leaves", which 
-        is identified by leaf rank (1 based ranking).
-        
-        
-        On commencing simulation
-        ------------------------
-        OnSimulationCommencing is called on commencing simulation. 
-        The leaves in the seed are initialized from all children with model 
-        \ref Models.PMF.Organs.LeafCohort "LeafCohort". 
-        
-        Frost impact on leaf death
-        ------------------------
-        Each cohort leaf is killed by a fraction if value of FrostFraction is more than 0. 
-        The frost fraction could be calculated through daily minimum temperature and
-        growth stages. See \ref Models.PMF.Organs.LeafCohort "LeafCohort" model for details.
-        See \subpage parameter "tutorial" about how to set up a parameter in APSIM. 
-        </remarks>
-    */
+    /// <summary>
+    /// A generic leaf model
+    /// </summary>
+    /// \param KDead The extinction coefficient for dead leaf (\f$k_d\f$).
+    /// \param FrostFraction The fraction of leaf death caused by frost event.
+    /// \retval MaxCover The maximum coverage (\f$C_{max}\f$) with default value 1, 
+    ///     which is set by manager sowing. 
+    /// \retval LAI Leaf area index for green leaf (\f$\text{LAI}_{g}\f$, \f$m^2 m^{-2}\f$)
+    /// \retval LAIDead Leaf area index for dead leaf  (\f$\text{LAI}_{d}\f$, \f$m^2 m^{-2}\f$)
+    /// \retval LAITotal Total LAI including live and dead parts (\f$m^2 m^{-2}\f$)
+    ///     \f[
+    /// /// LAI = \text{LAI}_{g} + \text{LAI}_{d}
+    ///     \f]
+    /// \retval CoverGreen Cover for green leaf (\f$C_g\f$, unitless). 
+    ///     \f$C_g\f$ is calculated according to
+    ///     extinction coefficient of green leaf (\f$k_{g}\f$).
+    ///     \f[
+    /// /// C_{g}=C_{max}(1-\exp(-k_{g}\frac{\text{LAI}_{g}}{C_{max}}))
+    ///     \f]
+    ///     where, \f$k\f$ is the extinction coefficient which calculates 
+    ///     by parameter "ExtinctionCoeff". 
+    ///     As the default value of \f$C_{max}\f$ is 1, the function is reduced to
+    ///     \f[
+    /// /// C_{g}=1-\exp(-k_{g}\text{LAI}_{g})
+    ///     \f]
+    /// \retval CoverDead Cover for dead leaf (\f$C_d\f$, unitless).
+    ///     \f$C_d\f$ is calculated according to
+    ///     extinction coefficient of dead leaf (\f$k_{d}\f$).
+    ///     \f[
+    /// /// C_{d}=1-\exp(-k_{d}\text{LAI}_{d})
+    ///     \f]
+    /// \retval CoverTotal Total cover for green and dead leaves (\f$C_t\f$, unitless).
+    ///     \f[
+    /// /// C_{t} = 1 - (1 - C_{g})(1 - C_{d})
+    ///     \f]
+    /// 
+    /// \retval Height Plant height (mm). 
+    /// \retval Depth Plant depth (mm). Equal to plant height (not sure its function?)
+    /// \retval FRGR Fractional relative growth rate (unitless, 0-1)
+    ///     with 1.0 at full growth rate and 0.0 at no growth.
+    /// \retval PotentialEP Potential evapotranspiration. Set by MICROCLIMATE.
+    /// 
+    /// <remarks>
+    /// The organ "Leaf" consists of a series of \ref LeafCohort "cohort leaves", which 
+    /// is identified by leaf rank (1 based ranking).
+    /// 
+    /// 
+    /// On commencing simulation
+    /// ------------------------
+    /// OnSimulationCommencing is called on commencing simulation. 
+    /// The leaves in the seed are initialized from all children with model 
+    /// \ref Models.PMF.Organs.LeafCohort "LeafCohort". 
+    /// 
+    /// Frost impact on leaf death
+    /// ------------------------
+    /// Each cohort leaf is killed by a fraction if value of FrostFraction is more than 0. 
+    /// The frost fraction could be calculated through daily minimum temperature and
+    /// growth stages. See \ref Models.PMF.Organs.LeafCohort "LeafCohort" model for details.
+    /// See \subpage parameter "tutorial" about how to set up a parameter in APSIM. 
+    /// </remarks>
     [Serializable]
     [Description("Leaf Class")]
     public class Leaf : BaseOrgan, AboveGround, ICanopy

--- a/Models/Plant/Organs/LeafCohort.cs
+++ b/Models/Plant/Organs/LeafCohort.cs
@@ -11,17 +11,15 @@ using APSIM.Shared.Utilities;
 
 namespace Models.PMF.Organs
 {
-    /*! <summary>
-        A leaf cohort model
-        </summary>
-        <remarks>
-        
-        Leaf death
-        ------------------------
-        The leaf area, structural biomass and structural nitrogen of 
-        green (live) parts is subtracted by a fraction.
-        </remarks>
-    */
+    ///<summary>
+    /// A leaf cohort model
+    /// </summary>
+    /// <remarks>
+    /// Leaf death
+    /// ------------------------
+    /// The leaf area, structural biomass and structural nitrogen of 
+    /// green (live) parts is subtracted by a fraction.
+    ///</remarks>
     [Serializable]
     public class LeafCohort : Model
     {

--- a/Models/Plant/Organs/Root.cs
+++ b/Models/Plant/Organs/Root.cs
@@ -11,52 +11,51 @@ using APSIM.Shared.Utilities;
 
 namespace Models.PMF.Organs
 {
-    /*! <summary>
-        The generic root model
-        </summary>
-        \param InitialDM <b>(Constant)</b> The initial dry weight of root (\f$g mm^{-2}\f$. CHECK).
-        \param SpecificRootLength <b>(Constant)</b> The length of the specific root 
-            (\f$m g^{-1}\f$. CHECK).
-        \param KNO3 <b>(Constant)</b> Fraction of extractable soil NO3 (\f$K_{NO3}\f$, unitless).  
-        \param KNH4 <b>(Constant)</b> Fraction of extractable soil NH4 (\f$K_{NH4}\f$, unitless).  
-        \param NitrogenDemandSwitch <b>(IFunction)</b> Whether to switch on nitrogen demand 
-            when nitrogen deficit is calculated (0 or 1, unitless).
-        \param RootFrontVelocity <b>(IFunction)</b> The daily growth speed of root depth 
-            (\f$mm d^{-1}\f$. CHECK).
-        \param PartitionFraction <b>(IFunction)</b> The fraction of biomass partitioning 
-            into root (0-1, unitless).
-        \param KLModifier <b>(IFunction)</b> The modifier for KL factor which is defined as 
-            the fraction of available water able to be extracted per day, and empirically 
-            derived incorporating both plant and soil factors which limit rate of water 
-            update (0-1, unitless).
-        \param TemperatureEffect <b>(IFunction)</b> 
-            The temperature effects on root depth growth (0-1, unitless).
-        \param MaximumNConc <b>(IFunction)</b> 
-            Maximum nitrogen concentration (\f$g m^{-2}\f$. CHECK).
-        \param MinimumNConc <b>(IFunction)</b> 
-            Minimum nitrogen concentration (\f$g m^{-2}\f$. CHECK).
-        \param MaxDailyNUptake <b>(IFunction)</b> 
-            Maximum daily nitrogen update (\f$kg ha^{-1}\f$. CHECK).
-        
-        \param SenescenceRate <b>(IFunction, Optional)</b> The daily senescence rate of 
-            root length (0-1, unitless).
-            
-        \retval Length Total root length (mm).
-        \retval Depth Root depth (mm).
-        
-        <remarks>
-        
-        Potential root growth 
-        ------------------------
-         
-        Actual root growth
-        ------------------------
-        
-        Nitrogen deficit 
-        ------------------------
-        
-        </remarks>
-    */
+    ///<summary>
+    /// The generic root model
+    ///</summary>
+    /// \param InitialDM <b>(Constant)</b> The initial dry weight of root (\f$g mm^{-2}\f$. CHECK).
+    /// \param SpecificRootLength <b>(Constant)</b> The length of the specific root 
+    ///     (\f$m g^{-1}\f$. CHECK).
+    /// \param KNO3 <b>(Constant)</b> Fraction of extractable soil NO3 (\f$K_{NO3}\f$, unitless).  
+    /// \param KNH4 <b>(Constant)</b> Fraction of extractable soil NH4 (\f$K_{NH4}\f$, unitless).  
+    /// \param NitrogenDemandSwitch <b>(IFunction)</b> Whether to switch on nitrogen demand 
+    ///     when nitrogen deficit is calculated (0 or 1, unitless).
+    /// \param RootFrontVelocity <b>(IFunction)</b> The daily growth speed of root depth 
+    ///     (\f$mm d^{-1}\f$. CHECK).
+    /// \param PartitionFraction <b>(IFunction)</b> The fraction of biomass partitioning 
+    ///     into root (0-1, unitless).
+    /// \param KLModifier <b>(IFunction)</b> The modifier for KL factor which is defined as 
+    ///     the fraction of available water able to be extracted per day, and empirically 
+    ///     derived incorporating both plant and soil factors which limit rate of water 
+    ///     update (0-1, unitless).
+    /// \param TemperatureEffect <b>(IFunction)</b> 
+    ///     The temperature effects on root depth growth (0-1, unitless).
+    /// \param MaximumNConc <b>(IFunction)</b> 
+    ///     Maximum nitrogen concentration (\f$g m^{-2}\f$. CHECK).
+    /// \param MinimumNConc <b>(IFunction)</b> 
+    ///     Minimum nitrogen concentration (\f$g m^{-2}\f$. CHECK).
+    /// \param MaxDailyNUptake <b>(IFunction)</b> 
+    ///     Maximum daily nitrogen update (\f$kg ha^{-1}\f$. CHECK).
+    /// 
+    /// \param SenescenceRate <b>(IFunction, Optional)</b> The daily senescence rate of 
+    ///     root length (0-1, unitless).
+    ///     
+    /// \retval Length Total root length (mm).
+    /// \retval Depth Root depth (mm).
+    /// 
+    ///<remarks>
+    /// 
+    /// Potential root growth 
+    /// ------------------------
+    ///  
+    /// Actual root growth
+    /// ------------------------
+    /// 
+    /// Nitrogen deficit 
+    /// ------------------------
+    /// 
+    ///</remarks>
     [Serializable]
     public class Root : BaseOrgan, BelowGround
     {

--- a/Models/Plant/Organs/SimpleLeaf.cs
+++ b/Models/Plant/Organs/SimpleLeaf.cs
@@ -11,31 +11,30 @@ using Models.Interfaces;
 
 namespace Models.PMF.Organs
 {
-    /*! <summary>
-        A simple leaf organ
-        </summary>
-        \retval LAI Leaf area index for green leaf (\f$\text{LAI}_{g}\f$, \f$m^2 m^{-2}\f$)
-        \retval LAIDead Leaf area index for dead leaf  (\f$\text{LAI}_{d}\f$, \f$m^2 m^{-2}\f$)
-        \retval LAITotal Total LAI including live and dead parts (\f$m^2 m^{-2}\f$)
-            \f[
-                LAI = \text{LAI}_{g} + \text{LAI}_{d}
-            \f]
-        \retval CoverGreen Cover for green leaf (\f$C_g\f$, unitless). The value of CoverFunction is returned 
-            if "CoverFunction" exists in the model. \f$C_g\f$ is calculated according to
-            extinction coefficient of green leaf (\f$k_{g}\f$) 
-            if "ExtinctionCoefficientFunction" exists in the model.
-            \f[
-                C_g = 1-\exp(-k_{g} * \text{LAI}_{g})
-            \f]
-            where, \f$k\f$ is the extinction coefficient which calculates by "ExtinctionCoefficientFunction"
-        \retval CoverDead Cover for dead leaf (\f$C_d\f$, unitless). \f$C_d\f$ is calculated according to 
-            extinction coefficient of dead leaf (\f$k_{d}\f$). 
-            \f[
-                C_d = 1-\exp(-k_{d} * \text{LAI}_{d})
-            \f]
-        <remarks>
-        </remarks>
-    */
+    /// <summary>
+    /// A simple leaf organ
+    /// </summary>
+    /// \retval LAI Leaf area index for green leaf (\f$\text{LAI}_{g}\f$, \f$m^2 m^{-2}\f$)
+    /// \retval LAIDead Leaf area index for dead leaf  (\f$\text{LAI}_{d}\f$, \f$m^2 m^{-2}\f$)
+    /// \retval LAITotal Total LAI including live and dead parts (\f$m^2 m^{-2}\f$)
+    ///     \f[
+    /// /// LAI = \text{LAI}_{g} + \text{LAI}_{d}
+    ///     \f]
+    /// \retval CoverGreen Cover for green leaf (\f$C_g\f$, unitless). The value of CoverFunction is returned 
+    ///     if "CoverFunction" exists in the model. \f$C_g\f$ is calculated according to
+    ///     extinction coefficient of green leaf (\f$k_{g}\f$) 
+    ///     if "ExtinctionCoefficientFunction" exists in the model.
+    ///     \f[
+    /// /// C_g = 1-\exp(-k_{g} * \text{LAI}_{g})
+    ///     \f]
+    ///     where, \f$k\f$ is the extinction coefficient which calculates by "ExtinctionCoefficientFunction"
+    /// \retval CoverDead Cover for dead leaf (\f$C_d\f$, unitless). \f$C_d\f$ is calculated according to 
+    ///     extinction coefficient of dead leaf (\f$k_{d}\f$). 
+    ///     \f[
+    /// /// C_d = 1-\exp(-k_{d} * \text{LAI}_{d})
+    ///     \f]
+    /// <remarks>
+    /// </remarks>
     [Serializable]
     public class SimpleLeaf : BaseOrgan, AboveGround, ICanopy
     {

--- a/Models/Plant/Phenology/VernalisationCW.cs
+++ b/Models/Plant/Phenology/VernalisationCW.cs
@@ -7,56 +7,54 @@ using System.Xml.Serialization;
 
 namespace Models.PMF.Phen
 {
-    
-    /*! <summary>
-        The vernalization and photoperiod effects from CERES wheat.
-        </summary>
-        \pre A \ref Models.PMF.Phen.Phenology "Phenology" model has to exist.
-        \pre A \ref Models.PMF.Functions.PhotoperiodFunction "PhotoperiodFunction" 
-            model has to exist to calculate the day length.
-        \pre A \ref Models.WeatherFile "WeatherFile" model has to exist to 
-        retrieve the daily minimum and maximum temperature.
-        \param VernSens The vernalization sensitivity
-        \param PhotopSens The photoperiod sensitivity
-        \param StartStageForEffects The start stage to calculate the vernalization and photoperiod effects
-        \param EndStageForEffects The end stage to calculate the vernalization and photoperiod effects
-        \param StartStageForCumulativeVD The start stage to calculate the cumulative vernalization
-        \param EndStageForCumulativeVD The end stage to calculate the cumulative vernalization
-        \retval VernEff The vernalization effects (from 0 to 1)
-        \retval PhotopEff The photoperiod effects (from 0 to 1)
-        <remarks>
-        Vernalization
-        ------------------------
-        In CERES-Wheat \cite jones_dssat_2003, vernalisation is simulated from daily average crown temperature (\f$T_{c}\f$), daily maximum (\f$T_{max}\f$) and 
-        minimum (\f$T_{min}\f$) temperatures using the original CERES approach.
-        \f[
-        \Delta V=\min(1.4-0.0778T_{c},\:0.5+13.44\frac{T_{c}}{(T_{max}-T_{min}+3)^{2}})\quad\text{when, }T_{max}<30\,{}^{\circ}\text{C}\:\text{and}\, T_{min}<15\,{}^{\circ}\text{C}
-        \f]
-        Devernalisation can occur if daily \f$T_{max}\f$ is above 30 \f$^{\circ}\text{C}\f$ and the total vernalisation (\f$V\f$) is less than 10 .
-        \f[
-        \Delta V_{d}=\min(0.5(T_{max}-30),\: V)\quad\text{when, }T_{max}>30\,{}^{\circ}\text{C}\;\text{and}\; V<10
-        \f]
-        The total vernalisation (\f$V\f$) is calculated by summing daily vernalisation and devernalisation from \p StartStageForEffects to \p EndStageForEffects. 
-        \f[
-        V=\sum(\Delta V-\Delta V_{d})
-        \f]
-        However, the vernalisation factor (\f$f_{v}\f$) is calculated just from \p StartStageForCumulativeVD to \p EndStageForCumulativeVD.
-        \f[
-        f_{V}=1-(0.0054545R_{V}+0.0003)\times(50-V)
-        \f]
 
-        Photoperiod
-        ------------------------
-        Photoperiod is calculated from day of year and latitude using standard astronomical equations accounting for civil twilight 
-        using the parameter \p twilight in \ref Models.PMF.Functions.PhotoperiodFunction. In APSIM, 
-        the photoperiod affects phenology between \p StartStageForEffects and \p EndStageForEffects. During this period, thermal time 
-        is affected by a photoperiod factor (\f$f_{D}\f$) that is calculated by
-        \f[
-        f_{D}=1-0.002R_{p}(20-L_{P})^{2}
-        \f]
-        where \f$L_{P}\f$ is the day length (h) from \ref Models.PMF.Functions.PhotoperiodFunction.
-        </remarks>
-    */
+    ///<summary>
+    /// The vernalization and photoperiod effects from CERES wheat.
+    ///</summary>
+    /// \pre A \ref Models.PMF.Phen.Phenology "Phenology" model has to exist.
+    /// \pre A \ref Models.PMF.Functions.PhotoperiodFunction "PhotoperiodFunction" 
+    ///     model has to exist to calculate the day length.
+    /// \pre A \ref Models.WeatherFile "WeatherFile" model has to exist to 
+    /// retrieve the daily minimum and maximum temperature.
+    /// \param VernSens The vernalization sensitivity
+    /// \param PhotopSens The photoperiod sensitivity
+    /// \param StartStageForEffects The start stage to calculate the vernalization and photoperiod effects
+    /// \param EndStageForEffects The end stage to calculate the vernalization and photoperiod effects
+    /// \param StartStageForCumulativeVD The start stage to calculate the cumulative vernalization
+    /// \param EndStageForCumulativeVD The end stage to calculate the cumulative vernalization
+    /// \retval VernEff The vernalization effects (from 0 to 1)
+    /// \retval PhotopEff The photoperiod effects (from 0 to 1)
+    ///<remarks>
+    /// Vernalization
+    /// ------------------------
+    /// In CERES-Wheat \cite jones_dssat_2003, vernalisation is simulated from daily average crown temperature (\f$T_{c}\f$), daily maximum (\f$T_{max}\f$) and 
+    /// minimum (\f$T_{min}\f$) temperatures using the original CERES approach.
+    /// \f[
+    /// \Delta V=\min(1.4-0.0778T_{c},\:0.5+13.44\frac{T_{c}}{(T_{max}-T_{min}+3)^{2}})\quad\text{when, }T_{max}&lt;30\,{}^{\circ}\text{C}\:\text{and}\, T_{min}&lt;15\,{}^{\circ}\text{C}
+    /// \f]
+    /// Devernalisation can occur if daily \f$T_{max}\f$ is above 30 \f$^{\circ}\text{C}\f$ and the total vernalisation (\f$V\f$) is less than 10 .
+    /// \f[
+    /// \Delta V_{d}=\min(0.5(T_{max}-30),\: V)\quad\text{when, }T_{max}&gt;30\,{}^{\circ}\text{C}\;\text{and}\; V&lt;10
+    /// \f]
+    /// The total vernalisation (\f$V\f$) is calculated by summing daily vernalisation and devernalisation from \p StartStageForEffects to \p EndStageForEffects. 
+    /// \f[
+    /// V=\sum(\Delta V-\Delta V_{d})
+    /// \f]
+    /// However, the vernalisation factor (\f$f_{v}\f$) is calculated just from \p StartStageForCumulativeVD to \p EndStageForCumulativeVD.
+    /// \f[
+    /// f_{V}=1-(0.0054545R_{V}+0.0003)\times(50-V)
+    /// \f]
+    /// Photoperiod
+    /// ------------------------
+    /// Photoperiod is calculated from day of year and latitude using standard astronomical equations accounting for civil twilight 
+    /// using the parameter \p twilight in \ref Models.PMF.Functions.PhotoperiodFunction. In APSIM, 
+    /// the photoperiod affects phenology between \p StartStageForEffects and \p EndStageForEffects. During this period, thermal time 
+    /// is affected by a photoperiod factor (\f$f_{D}\f$) that is calculated by
+    /// \f[
+    /// f_{D}=1-0.002R_{p}(20-L_{P})^{2}
+    /// \f]
+    /// where \f$L_{P}\f$ is the day length (h) from \ref Models.PMF.Functions.PhotoperiodFunction.
+    ///</remarks>
     [Serializable]
     public class VernalisationCW : Model
     {

--- a/Models/Plant/Plant.cs
+++ b/Models/Plant/Plant.cs
@@ -15,42 +15,43 @@ namespace Models.PMF
     using Models.PMF.Phen;
     using Models.Soils.Arbitrator;
 
-    /*! <summary>
-        The generic plant model
-        </summary>
-        \pre Summary A Summary model has to exist to write summary message.
-        \pre Phenology A \ref Models.PMF.Phen.Phenology Phenology model is 
-        optional to check whether plant has emerged.
-        \pre OrganArbitrator A OrganArbitrator model is optional (not used currently).
-        \pre Structure A Structure model is optional (not used currently).
-        \pre Leaf A \ref Models.PMF.Organs.Leaf Leaf model is optional 
-        to calculate water supply and demand ratio.
-        \pre Root A Models.PMF.Organs.Root Root model is optional 
-        to calculate water supply and demand ratio.
-        \param CropType Used by several organs to determine the type of crop.
-        \retval Population Number of plants per square meter. 
-        \retval IsAlive Return true if plant is alive and in the ground.
-        \retval IsEmerged Return true if plant has emerged.
-        
-        On commencing simulation
-        ------------------------
-        OnSimulationCommencing is called on commencing simulation. Organs contain 
-        all children which derive from model IOrgan. The model variables 
-        are reset.
-        
-        On sowing 
-        -------------------------
-        Plant is sown by a manager script in a APSIM model. For example,    
-        \code
-        2012-10-23 [Maize].Sow(population:11, cultivar:"Pioneer_3153", depth:50, rowSpacing:710);
-        \endcode
-        
-        Sowing parameters should be specified, i.e. cultivar, population, depth, rowSpacing,
-        maxCover (optional), and budNumber (optional).
-        
-        Two events "Sowing" and "PlantSowing" are invoked to notify other models 
-        to execute sowing events.
-    */
+    ///<summary>
+    /// The generic plant model
+    /// </summary>
+    /// \pre Summary A Summary model has to exist to write summary message.
+    /// \pre Phenology A \ref Models.PMF.Phen.Phenology Phenology model is 
+    /// optional to check whether plant has emerged.
+    /// \pre OrganArbitrator A OrganArbitrator model is optional (not used currently).
+    /// \pre Structure A Structure model is optional (not used currently).
+    /// \pre Leaf A \ref Models.PMF.Organs.Leaf Leaf model is optional 
+    /// to calculate water supply and demand ratio.
+    /// \pre Root A Models.PMF.Organs.Root Root model is optional 
+    /// to calculate water supply and demand ratio.
+    /// \param CropType Used by several organs to determine the type of crop.
+    /// \retval Population Number of plants per square meter. 
+    /// \retval IsAlive Return true if plant is alive and in the ground.
+    /// \retval IsEmerged Return true if plant has emerged.
+    /// 
+    /// On commencing simulation
+    /// ------------------------
+    /// OnSimulationCommencing is called on commencing simulation. Organs contain 
+    /// all children which derive from model IOrgan. The model variables 
+    /// are reset.
+    /// 
+    /// On sowing 
+    /// -------------------------
+    /// Plant is sown by a manager script in a APSIM model. For example,    
+    /// \code
+    /// 2012-10-23 [Maize].Sow(population:11, cultivar:"Pioneer_3153", depth:50, rowSpacing:710);
+    /// \endcode
+    /// 
+    /// Sowing parameters should be specified, i.e. cultivar, population, depth, rowSpacing,
+    /// maxCover (optional), and budNumber (optional).
+    /// 
+    /// Two events "Sowing" and "PlantSowing" are invoked to notify other models 
+    /// to execute sowing events.
+    /// <remarks>
+    /// </remarks>
     [Serializable]
     public class Plant : ModelCollectionFromResource, ICrop
     {

--- a/Models/Plant/Structure.cs
+++ b/Models/Plant/Structure.cs
@@ -10,139 +10,137 @@ using Models.PMF.Functions.StructureFunctions;
 
 namespace Models.PMF
 {
-    /*!
-        <summary>
-        A structure model for plant
-        </summary>
-        \pre A \ref Models.PMF.Plant "Plant" model has to exist to access 
-            sowing data, e.g. population, bud number.
-        \pre A \ref Models.PMF.Organs.Leaf "Leaf" model has to exist to 
-            access leaf initialisation an number.
-        \pre A \ref Models.PMF.Phen.Phenology "Phenology" model has to exist 
-            to check whether plant is initialised.
-        \param InitialiseStage <b>(Constant)</b> The initialise stage. 
-            The primordia number, node number and stem population start to 
-            increase since initialise stage.
-        \param PrimaryBudNo <b>(Constant)</b> The primary bud number in each node. 
-            However, PrimaryBudNo is reset at sowing using sowing script 
-            with default value 1.
-            
-        \param ThermalTime <b>(IFunction)</b> The daily thermal time (<sup>o</sup>Cd).
-        \param MainStemPrimordiaInitiationRate <b>(IFunction)</b> The initiation rate 
-            of primordia in main stem (# <sup>o</sup>Cd<sup>-1</sup>).
-        \param MainStemNodeAppearanceRate <b>(IFunction)</b> The appearance rate 
-            of node in main stem (# <sup>o</sup>Cd<sup>-1</sup>).
-        \param MainStemFinalNodeNumber <b>(IFunction)</b> The maximum node number
-            in main stem (#). The maximum node is calculated by model 
-            MainStemFinalNodeNumberFunction if MainStemFinalNodeNumber is 
-            specified to MainStemFinalNodeNumberFunction.
-        \param HeightModel <b>(IFunction)</b> Plant height (mm).
-        \param BranchingRate <b>(IFunction)</b> The rate of new branching (#).
-        \param ShadeInducedBranchMortality <b>(IFunction)</b> 
-            The branch mortality induced by shading (0-1).
-        \param DroughtInducedBranchMortality <b>(IFunction)</b> 
-            The branch mortality induced by drought (0-1).
-        \param PlantMortality <b>(IFunction, Optional)</b> The plant mortality 
-            to reduce plant population.
-            
-        \retval TotalStemPopn Number of stems including main and branch stems 
-            (m<sup>-2</sup>).
-        \retval MainStemPrimordiaNo Number of primordia initiated in main stem.
-        \retval MainStemNodeNo Number of nodes appeared in main stem.
-        \retval PlantTotalNodeNo Number of leaves appeared per plant including 
-            all main stem and branch leaves.
-        \retval ProportionBranchMortality Proportion of branch mortality.
-        \retval ProportionPlantMortality Proportion of plant mortality.
-        \retval DeltaNodeNumber The daily changes of node number.
-        
-        \retval MainStemPopn Number of main stem per square meter (m<sup>-2</sup>).
-        \retval RemainingNodeNo Number of leaves yet to appear.
-        \retval Height Plant height (mm).
-        \retval PrimaryBudTotalNodeNo Number of appeared bud (leaves) per primary 
-            bud unit including all main stem and branch leaves.
-        \retval MainStemFinalNodeNo The main stem final node number.
-        \retval RelativeNodeApperance Relative progress toward final leaf (0-1).
-        
-        <remarks>
-        The Structure model predicts the development and mortality of primordia, node (leaf) and branch
-        (tiller). The primary bud number assumes the same in seed and all axillary bud and set in the 
-        sowing data with default value 1 (Not sure the assumption is correct).
-        
-        ## Simulation commencing (OnSimulationCommencing)
-            Local variables are reset to 0 including total stem population, number of 
-            main stem primordia, main stem node and plant total node;
-            proportion of branch and plant mortality, changes of node number.
-            
-        ## Sowing (OnPlantSowing)
-            The primary bud number and population of main stem are initialised from 
-            sowing date. Primary bud number equals to bud number bud number at 
-            sowing data (default value equals to 1). The population of main stem
-            equals to bud number multiplying sowing population.
-            
-            The maximum node (leaf) number in main stem is calculated 
-                by MainStemFinalNodeNumber. 
-                \ref Models.PMF.Functions.StructureFunctions.MainStemFinalNodeNumberFunction "MainStemFinalNodeNumberFunction" is called if 
-                MainStemFinalNodeNumber specifies to MainStemFinalNodeNumberFunction.
-                
-            Plant height is calculated by HeightModel.
-            
-        ## Plant ending (OnPlantEnding)
-            See simulation commencing
-            
-        ## Potential growth (OnDoPotentialPlantGrowth)
-            Daily changes of primordia (\f$\Delta N_{p}\f$) and node (leaf) (\f$\Delta N_{n}\f$) 
-            number in main stem are calculated using plastochron (\f$P_{pla}\f$) 
-            and phyllochron (\f$P_{phy}\f$), respectively, after plant initialisation.
-            
-            \f[
-                \Delta N_{p}=\frac{\Delta TT_{d}}{P_{pla}}
-            \f]
-            
-            \f[
-                \Delta N_{n}=\frac{\Delta TT_{d}}{P_{phy}}
-            \f]
-            where, \f$\Delta TT_{d}\f$ is the daily thermal time, which calculates 
-            from parameter ThermalTime. \f$P_{pla}\f$ and \f$P_{phy}\f$ calculates
-            from parameter MainStemPrimordiaInitiationRate and MainStemNodeAppearanceRate,
-            respectively.
-            
-            Total number of primordia \f$N_{p}\f$ and node \f$N_{n}\f$ in main stem are summarised since initialisation.
-            \f[
-                N_{p}=\sum_{t=T_{i}}^{T}\Delta N_{p}
-            \f]
-            
-            \f[
-                N_{n}=\sum_{t=T_{i}}^{T}\Delta N_{n}
-            \f]
-            where, \f$T_{i}\f$ is day of plant initialisation. \f$T\f$ is today.
-            
-            Plant population (\f$P\f$) is daily reduced by the plant mortality (\f$\Delta P\f$).
-            \f[
-                P=P_{0}\prod_{t=T_{i}}^{T}(1-\Delta P)
-            \f]
-            where, \f$P_{0}\f$ is the sown population, which initialised at sowing.
-            
-            Total stem (main stem + branching) number (\f$N_{s}\f$) is initialised 
-            according plant population (\f$P\f$) and primary bud number (\f$N_b\f$).
-            \f[
-                N_{s}=P \times N_b
-            \f]
-            The total stem number is calculated by the daily increase of branching 
-            (\f$\Delta N_{s}\f$), and daily decrease caused by drought 
-            (\f$\Delta N_{drought}\f$) and shade mortality (\f$\Delta N_{shade}\f$).
-            \f[
-            N_{s}=PN_{b}[1+\sum_{t=t_{i}}^{T}\Delta N_{s}+\prod_{t=T_{i}}^{T}(1-\Delta N_{drought}-\Delta N_{shade})]
-            \f]
-            Daily increase of branching (\f$\Delta N_{s}\f$) is calculated by parameter
-            BranchingRate. The mortalities, caused by drought (\f$\Delta N_{drought}\f$)  
-            and shade (\f$\Delta N_{shade}\f$) are calculated by parameters 
-            DroughtInducedBranchMortality and ShadeInducedBranchMortality, respectively.
-            
-        ## Actual growth (OnDoActualPlantGrowth)
-            Plant total node number is updated according to appeared node (leaf) number 
-            and population.
-        </remarks>
-    */
+    /// <summary>
+    /// A structure model for plant
+    /// </summary>
+    /// \pre A \ref Models.PMF.Plant "Plant" model has to exist to access 
+    /// sowing data, e.g. population, bud number.
+    /// \pre A \ref Models.PMF.Organs.Leaf "Leaf" model has to exist to 
+    /// access leaf initialisation an number.
+    /// \pre A \ref Models.PMF.Phen.Phenology "Phenology" model has to exist 
+    /// to check whether plant is initialised.
+    /// \param InitialiseStage <b>(Constant)</b> The initialise stage. 
+    /// The primordia number, node number and stem population start to 
+    /// increase since initialise stage.
+    /// \param PrimaryBudNo <b>(Constant)</b> The primary bud number in each node. 
+    /// However, PrimaryBudNo is reset at sowing using sowing script 
+    /// with default value 1.
+    /// 
+    /// \param ThermalTime <b>(IFunction)</b> The daily thermal time (<sup>o</sup>Cd).
+    /// \param MainStemPrimordiaInitiationRate <b>(IFunction)</b> The initiation rate 
+    /// of primordia in main stem (# <sup>o</sup>Cd<sup>-1</sup>).
+    /// \param MainStemNodeAppearanceRate <b>(IFunction)</b> The appearance rate 
+    /// of node in main stem (# <sup>o</sup>Cd<sup>-1</sup>).
+    /// \param MainStemFinalNodeNumber <b>(IFunction)</b> The maximum node number
+    /// in main stem (#). The maximum node is calculated by model 
+    /// MainStemFinalNodeNumberFunction if MainStemFinalNodeNumber is 
+    /// specified to MainStemFinalNodeNumberFunction.
+    /// \param HeightModel <b>(IFunction)</b> Plant height (mm).
+    /// \param BranchingRate <b>(IFunction)</b> The rate of new branching (#).
+    /// \param ShadeInducedBranchMortality <b>(IFunction)</b> 
+    /// The branch mortality induced by shading (0-1).
+    /// \param DroughtInducedBranchMortality <b>(IFunction)</b> 
+    /// The branch mortality induced by drought (0-1).
+    /// \param PlantMortality <b>(IFunction, Optional)</b> The plant mortality 
+    /// to reduce plant population.
+    /// 
+    /// \retval TotalStemPopn Number of stems including main and branch stems 
+    /// (m<sup>-2</sup>).
+    /// \retval MainStemPrimordiaNo Number of primordia initiated in main stem.
+    /// \retval MainStemNodeNo Number of nodes appeared in main stem.
+    /// \retval PlantTotalNodeNo Number of leaves appeared per plant including 
+    /// all main stem and branch leaves.
+    /// \retval ProportionBranchMortality Proportion of branch mortality.
+    /// \retval ProportionPlantMortality Proportion of plant mortality.
+    /// \retval DeltaNodeNumber The daily changes of node number.
+    /// 
+    /// \retval MainStemPopn Number of main stem per square meter (m<sup>-2</sup>).
+    /// \retval RemainingNodeNo Number of leaves yet to appear.
+    /// \retval Height Plant height (mm).
+    /// \retval PrimaryBudTotalNodeNo Number of appeared bud (leaves) per primary 
+    /// bud unit including all main stem and branch leaves.
+    /// \retval MainStemFinalNodeNo The main stem final node number.
+    /// \retval RelativeNodeApperance Relative progress toward final leaf (0-1).
+    /// 
+    /// <remarks>
+    /// The Structure model predicts the development and mortality of primordia, node (leaf) and branch
+    /// (tiller). The primary bud number assumes the same in seed and all axillary bud and set in the 
+    /// sowing data with default value 1 (Not sure the assumption is correct).
+    /// 
+    /// ## Simulation commencing (OnSimulationCommencing)
+    /// Local variables are reset to 0 including total stem population, number of 
+    /// main stem primordia, main stem node and plant total node;
+    /// proportion of branch and plant mortality, changes of node number.
+    /// 
+    /// ## Sowing (OnPlantSowing)
+    /// The primary bud number and population of main stem are initialised from 
+    /// sowing date. Primary bud number equals to bud number bud number at 
+    /// sowing data (default value equals to 1). The population of main stem
+    /// equals to bud number multiplying sowing population.
+    /// 
+    /// The maximum node (leaf) number in main stem is calculated 
+    ///     by MainStemFinalNodeNumber. 
+    ///     \ref Models.PMF.Functions.StructureFunctions.MainStemFinalNodeNumberFunction "MainStemFinalNodeNumberFunction" is called if 
+    ///     MainStemFinalNodeNumber specifies to MainStemFinalNodeNumberFunction.
+    ///     
+    /// Plant height is calculated by HeightModel.
+    /// 
+    /// ## Plant ending (OnPlantEnding)
+    /// See simulation commencing
+    /// 
+    /// ## Potential growth (OnDoPotentialPlantGrowth)
+    /// Daily changes of primordia (\f$\Delta N_{p}\f$) and node (leaf) (\f$\Delta N_{n}\f$) 
+    /// number in main stem are calculated using plastochron (\f$P_{pla}\f$) 
+    /// and phyllochron (\f$P_{phy}\f$), respectively, after plant initialisation.
+    /// 
+    /// \f[
+    ///     \Delta N_{p}=\frac{\Delta TT_{d}}{P_{pla}}
+    /// \f]
+    /// 
+    /// \f[
+    ///     \Delta N_{n}=\frac{\Delta TT_{d}}{P_{phy}}
+    /// \f]
+    /// where, \f$\Delta TT_{d}\f$ is the daily thermal time, which calculates 
+    /// from parameter ThermalTime. \f$P_{pla}\f$ and \f$P_{phy}\f$ calculates
+    /// from parameter MainStemPrimordiaInitiationRate and MainStemNodeAppearanceRate,
+    /// respectively.
+    /// 
+    /// Total number of primordia \f$N_{p}\f$ and node \f$N_{n}\f$ in main stem are summarised since initialisation.
+    /// \f[
+    ///     N_{p}=\sum_{t=T_{i}}^{T}\Delta N_{p}
+    /// \f]
+    /// 
+    /// \f[
+    ///     N_{n}=\sum_{t=T_{i}}^{T}\Delta N_{n}
+    /// \f]
+    /// where, \f$T_{i}\f$ is day of plant initialisation. \f$T\f$ is today.
+    /// 
+    /// Plant population (\f$P\f$) is daily reduced by the plant mortality (\f$\Delta P\f$).
+    /// \f[
+    ///     P=P_{0}\prod_{t=T_{i}}^{T}(1-\Delta P)
+    /// \f]
+    /// where, \f$P_{0}\f$ is the sown population, which initialised at sowing.
+    /// 
+    /// Total stem (main stem + branching) number (\f$N_{s}\f$) is initialised 
+    /// according plant population (\f$P\f$) and primary bud number (\f$N_b\f$).
+    /// \f[
+    ///     N_{s}=P \times N_b
+    /// \f]
+    /// The total stem number is calculated by the daily increase of branching 
+    /// (\f$\Delta N_{s}\f$), and daily decrease caused by drought 
+    /// (\f$\Delta N_{drought}\f$) and shade mortality (\f$\Delta N_{shade}\f$).
+    /// \f[
+    /// N_{s}=PN_{b}[1+\sum_{t=t_{i}}^{T}\Delta N_{s}+\prod_{t=T_{i}}^{T}(1-\Delta N_{drought}-\Delta N_{shade})]
+    /// \f]
+    /// Daily increase of branching (\f$\Delta N_{s}\f$) is calculated by parameter
+    /// BranchingRate. The mortalities, caused by drought (\f$\Delta N_{drought}\f$)  
+    /// and shade (\f$\Delta N_{shade}\f$) are calculated by parameters 
+    /// DroughtInducedBranchMortality and ShadeInducedBranchMortality, respectively.
+    /// 
+    /// ## Actual growth (OnDoActualPlantGrowth)
+    /// Plant total node number is updated according to appeared node (leaf) number 
+    /// and population.
+    /// </remarks>
     [Serializable]
     [Description("Keeps Track of Plants Structural Development")]
     public class Structure : Model

--- a/Models/Weather.cs
+++ b/Models/Weather.cs
@@ -13,23 +13,22 @@ namespace Models
     using Models.Core;
     using APSIM.Shared.Utilities;
 
-    /*! <summary>
-        Reads in weather data and makes it available to other models.
-        </summary>
-        
-        <remarks>
-        Keywords in the weather file
-        - Maxt for maximum temperature
-        - Mint for minimum temperature
-        - Radn for radiation
-        - Rain for rainfall
-        - VP for vapour pressure
-        - Wind for wind speed
-        
-        VP is calculated using function Utility.Met.svp
-        Wind assign default value 3 if Wind is not resent.
-        </remarks>
-    */
+    ///<summary>
+    /// Reads in weather data and makes it available to other models.
+    ///</summary>
+    ///    
+    ///<remarks>
+    /// Keywords in the weather file
+    /// - Maxt for maximum temperature
+    /// - Mint for minimum temperature
+    /// - Radn for radiation
+    /// - Rain for rainfall
+    /// - VP for vapour pressure
+    /// - Wind for wind speed
+    ///     
+    /// VP is calculated using function Utility.Met.svp
+    /// Wind assign default value 3 if Wind is not resent.
+    ///</remarks>
     [Serializable]
     [ViewName("UserInterface.Views.TabbedMetDataView")]
     [PresenterName("UserInterface.Presenters.MetDataPresenter")]

